Makefile_packaging.mk

```diff
@@ -384,7 +384,12 @@ ifndef DEBFULLNAME
 endif
 
 test:
-	@echo "No test defined for this module"
+	# Test the rpmbuild by installing the built RPM
+	rm -f /etc/yum.repos.d/*"${DAOS_STACK_LOCAL_REPO//\//_}"
+	yum-config-manager --add-repo="$REPOSITORY_URL"/"$DAOS_STACK_LOCAL_REPO"
+	echo "gpgcheck = False" >> /etc/yum.repos.d/*"${DAOS_STACK_LOCAL_REPO//\//_}".repo
+	$(call install_repos,$(NAME)@$(BRANCH_NAME):$(BUILD_NUMBER))
+	yum -y install $(NAME)
```

Comment:
I think this should be included in `install_repos` so that other test writers that want to provide a more comprehensive test can get the benefit of the repos without having to reproduce all of the code.  Or maybe installing the group and local repos should be another function.  Your call.

But I've also misled you here.  The variable containing the repo isn't `$DAOS_STACK_LOCAL_REPO`.  That's a local variable defined in node provisioning from the per-distro Jenkins global variable.  You can see how to access those Jenkins per-distro repo global variables here: https://github.com/daos-stack/packaging/blob/697375ea812c8c84e03d7c4b597a019d4d8c3b95/Makefile_packaging.mk#L261-L271:

Both the distro _LOCAL_ and _GROUP_ repos should be installed.

Reply:
If I understand your comment correctly, are you saying that `install_repos` should look like this?
```
define install_repos
	for repo in $($(DISTRO_BASE)_PR_REPOS)                              \
	            $(DAOS_STACK_$(DISTRO_BASE)_LOCAL_REPO)                 \
	            $(DAOS_STACK_$(DISTRO_BASE)_GROUP_REPO)                 \
	            $(PR_REPOS) $(1); do                                    \
        ...
```
Or given that `$(DAOS_STACK_$(DISTRO_BASE)_GROUP_REPO)` or `$(DAOS_STACK_$(DISTRO_BASE)_LOCAL_REPO)` may not be defined, `$(DISTRO_BASE)_LOCAL_REPOS` should be used instead:
```
define install_repos
	for repo in $($(DISTRO_BASE)_PR_REPOS)                              \
	            $(DISTRO_BASE)_LOCAL_REPOS                              \
	            $(PR_REPOS) $(1); do                                    \
        ...
```

And then `test:` should simply be:
```
test:
	# Test the rpmbuild by installing the built RPM
	$(call install_repos,$(NAME)@$(BRANCH_NAME):$(BUILD_NUMBER))
	yum -y install $(NAME)
```
Or am I missing something?

Reply:
Yeah, that second option looks more correct, re-using the existing code for determining the global repos and the `test:` recipe looks about right.